addons/event/models/event_registration.py

```diff
@@ -128,7 +128,10 @@ def _synchronize_partner_values(self, partner, fnames=None):
     @api.model_create_multi
     def create(self, vals_list):
         registrations = super(EventRegistration, self).create(vals_list)
-
+        registrations_to_check_for_seats = registrations.filtered(
+            lambda reg: reg.event_id.auto_confirm if reg.state == "draft" else reg.state != "canceled"
+        )
+        registrations_to_check_for_seats._check_if_seats_available_before_confirming()
```

Comment:
``_check_auto_confirmation`` already checks for available seats. Is there any change we want here ? There are a lot of check methods called. This makes code difficult to follow and understand ... seems we should try to keep a simply contraints / check flow at one or two entry points (event and ticket ?), and avoid multiplying calls to check methods.